waiter/src/waiter/token.clj

```diff
@@ -236,15 +236,11 @@
           (let [token-owner (get token-metadata "owner")
                 version-hash (get headers "if-match")]
             (if hard-delete
-              (do
-                (when-not version-hash
```

Comment:
Are we still ensuring that hard-deletes aren't happening on non-soft-deleted tokens?

Reply:
Did we used to ensure it in the past? Afaik, any admin can hard-delete any token. 
The token syncer does hard deletes only if the token is soft-deleted on all clusters.

Reply:
What allows us to remove this constraint?  Why don't we want `If-Match` on hard-deletes?

Reply:
If we are hard deleting, require `If-Match` if not soft-deleted.